### DISCUSSION TITLE
Event Listener, is it always listening?

### DISCUSSION THREAD

**Discussion by bvinson23** [nss-day-cohort-47/C47-Discussions#40] [Help Me] [answered]
Brandon Vinson

There are no other tickets with my issue.

## Describe your issue
I've added an event listener to my main.js for my journal with the query selector of ".dailyjournal" which is the class I have attached to my main tag. I can click below the h3 tag and above the p tag that contains my ${entry.date} and I will get the console.log that I've setup, but if I click on the actual text of the entry, nothing happens. If I click below the ${entry.date} I get a console.log of undefined.

## Any error messages?
No error messages

## What I've googled
https://stackoverflow.com/questions/22238722/event-listener-not-working

## What have you tried?
I tried adding an event listener just to anywhere on the page and that doesn't work at all. So I find it odd that I can get a more specific event listener to work, even if a little.

## Link to the files in my repo
https://github.com/bvinson23/daily-journal/blob/bv-json/src/scripts/main.js
https://github.com/bvinson23/daily-journal/blob/bv-json/src/index.html

**Comment by BryanNilsen** [comment, marked as answer]
I see this condition on the event listener - `if (event.target.id.startsWith("entry"))`
Check to see if the elements you're clicking on (the "target") have an id that starts with "entry".
If not, you won't see the message in the console.

**Reply by bvinson23** [reply]
My section id starts with entry, so wouldn't anything within that section recognize the listener?

**Reply by BryanNilsen** [reply]
you're specifically referencing the "target", which is the actual element the event is triggered on. So even if you have the listener on the whole section and the p tag is within that section, the event.target is the p tag. If that p tag doesn't have an id that starts with "whatever" then that condition is not met and the code inside that condition won't run

**Reply by BryanNilsen** [reply]
to visualize this better... throw a console log before the condition and explore the event object to see what the event.target.id is
```
applicationElement.addEventListener("click", event => {
    console.log("This is the event", event)
    if (event.target.id.startsWith("entry")) {
        console.log("post clicked", event.target.id.split("--"))
        console.log("the id is", event.target.id.split("--")[1])
    }
})
```

**Reply by bvinson23** [reply]
❗ That's what I'm doing right now!! Thanks! I'll let you know what I find!

**Reply by bvinson23** [reply]
The console log showed me the exact id of the target. Understanding that I typically won't have an event listener on the entire post made it make more sense.
